test/path.jl

```diff
@@ -31,15 +31,33 @@
         @test isdirpath(S(".."))
     end
     @testset "joinpath" begin
+        @test joinpath(S("")) == ""
         @test joinpath(S("foo")) == "foo"
         @test joinpath(S("foo"), S("bar")) == "foo$(sep)bar"
+        @test joinpath(S("foo"), S("bar"), S("baz")) == "foo$(sep)bar$(sep)baz"
+        @test joinpath(S("foo"), S(""), S("baz")) == "foo$(sep)baz"
+        @test joinpath(S("foo"), S(""), S("")) == "foo$(sep)"
+        @test joinpath(S("foo"), S(""), S(""), S("bar")) == "foo$(sep)bar"
+
         @test joinpath(S("foo"), S(homedir())) == homedir()
         @test joinpath(S(abspath("foo")), S(homedir())) == homedir()
 
         if Sys.iswindows()
             @test joinpath(S("foo"),S("bar:baz")) == "bar:baz"
             @test joinpath(S("C:"),S("foo"),S("D:"),S("bar")) == "D:bar"
             @test joinpath(S("C:"),S("foo"),S("D:bar"),S("baz")) == "D:bar$(sep)baz"
+
+            # relative folders and case-insensitive drive letters
+            @test joinpath(S("C:\\a\\b"), S("c:c\\e")) == "c:\\a\\b\\c\\e"
```

Comment:
Should the earlier or later capitalization of the drive name be kept?

Reply:
The preference is to keep the later capitalization. This follows the general rule to prefer the  path  of `p2`  in `joinpath(p1,p2)`, or if `p2` is absolute to override `p1` completely.

Reply:
What does Python do here? I would think that in the case where `a` is absolute and `b` is relative, if they are joinable (i.e. the drives are compatible) one would want `joinpath(a, b)` to start with `a` and end with `b`.

Reply:
> What does Python do here? I


Python keeps the later capitalization.  But it doesn't really matter, since it is case-insentive.

Reply:
That's interesting. I was thinking that this might affect the definition of `relpath` but that doesn't seem to depend on `joinpath` at all in its logic, although I suspect it probably does have a bug in the case of drive letters with case mismatches.

Reply:
I see, honestly I haven't had a close look at `relpath` to tell how this affects `relpath`. My general thinking is that it would be a good idea to audit all of our FileSystem API, to check how we do with UNC paths. We also don't support extended paths, which is another good extension in the future.

Reply:
```suggestion
            @test joinpath(S("C:\\a\\b"), S("c:c\\e")) == "C:\\a\\b\\c\\e"
```
And change this test.